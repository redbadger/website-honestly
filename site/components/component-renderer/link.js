// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
