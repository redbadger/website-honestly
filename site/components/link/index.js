// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
