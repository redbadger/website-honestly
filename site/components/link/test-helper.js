// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
