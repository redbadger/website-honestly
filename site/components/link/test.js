// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
