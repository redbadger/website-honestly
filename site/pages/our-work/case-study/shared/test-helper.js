// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
