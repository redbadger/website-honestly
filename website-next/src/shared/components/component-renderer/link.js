import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
