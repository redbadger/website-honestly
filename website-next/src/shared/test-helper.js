import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
