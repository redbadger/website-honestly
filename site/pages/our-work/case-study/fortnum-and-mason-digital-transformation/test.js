import React from 'react';
import { mount } from 'enzyme';

import { Context } from '../../../../components/link/test-helper';
import MockedWhatToReadNext from '../shared/test-helper';
import FMTeaCaseStudy from './index';

jest.mock('../shared/what-to-read-next', () => MockedWhatToReadNext);

describe('site/pages/our-work/case-study/fortnum-and-mason-digital-transformation/', () => {
  it('renders correctly', () => {
    expect(
      mount(
        <Context>
          <FMTeaCaseStudy />
        </Context>,
      ),
    ).toMatchSnapshot();
  });
});
