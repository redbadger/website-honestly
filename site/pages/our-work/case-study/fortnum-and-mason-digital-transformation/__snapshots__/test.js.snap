// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/our-work/case-study/fortnum-and-mason-digital-transformation/ renders correctly 1`] = `
Array [
  <div
    class="header"
  >
    <picture
      class="picture"
    >
      <source
        media="largeScreen"
        srcset="header-l.jpg"
      />
      <source
        media="mediumScreen"
        srcset="header-m.jpg"
      />
      <img
        alt="A box of Fortnum & Mason tea"
        class="img header__image"
        srcset="header-s.jpg"
      />
    </picture>
  </div>,
  <div
    class="content"
  >
    <div
      class="content__wrapper"
    >
      <div
        class="content__mainTitleWrapper"
      >
        <h1
          class="content__mainTitle"
        >
          The ongoing digital transformation of a 310-year old retailer
        </h1>
      </div>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Let's make things better.
        </span>
        Make everyday special
      </h2>
      <p
        class="content__paragraph"
      >
        Since 2014, we’ve been working with Fortnum & Mason to drive business growth through digital transformation via an ongoing programme of tech initiatives across the website that helps make everyday special for their customers across all touch points. You can read about our award-winning work on the website 
        <a
          class="link"
          href="/our-work/case-study/fortnum-and-mason/"
          rel="noopener noreferrer"
          target="_blank"
        >
          here
        </a>
        .
      </p>
      <p
        class="content__paragraph"
      >
        Since then, the focus has been on increasing the awareness and accessibility of a wide array of Fortnum’s products; and the need to improve one particular category, tea.
      </p>
      <p
        class="content__paragraph"
      >
        Fortnum & Mason is famous across the globe for its dedication to supplying the finest quality selection of tea however the online experience needed more visibility to drive interest and revenue, as well as share the impressive breadth with online customers in a new, innovative way. In essence, we needed to bring the gifting and curatorial elements that exist with hampers, to the tea consumer journey.
      </p>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Do the right thing. Do the thing right.
        </span>
        Giving tea a digital twist and improving online experience
      </h2>
      <p
        class="content__paragraph"
      >
        The solution was the Tea Experience - a design to help improve the browsing experience and enable customers find products faster and with ease to enjoy the wide array of teas on offer outside of the store’s walls.
      </p>
      <div
        class="contactBox"
      >
        <h2
          class="contactBox__heading"
        >
          Project in mind?
        </h2>
        <a
          class="contactBox__button"
          href="#contactUs"
        >
          Tell us more
        </a>
      </div>
      <p
        class="content__paragraph"
      >
        We implemented advanced search capabilities so that the Tea Experience could be a fast, intuitive online shopping platform that interacted with Fortnum’s online store in a slick, interactive process.
      </p>
      <p
        class="content__paragraph"
      >
        In order to make this a reality, we integrated Fortnum’s online store with Elasticsearch, a search engine that allows structured, unstructured, geo and metric search structures, which deliver a quicker and smarter customer browsing experience. Whatever the customer query or search intention, the intelligent platform now allows customers to get a genuine answer to their questions, just as they would in store. On top of this, enhancements including intuitive faceted filtering has improved customer experience further and helped to drive conversation rates upwards.
      </p>
      <p
        class="content__paragraph"
      >
        Alongside implementing the likes of the Tea Experience, we ensure the smooth running of Fortnum’s website every day and the work has resulted in the website having its biggest sales day, week and month in the last quarter of 2016.
      </p>
      <p
        class="content__paragraph"
      >
        The team deploys up to five releases per day, getting new features and enhancements in front of customers quickly and enabling Fortnum & Mason to realise revenue increases instantly.
      </p>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Creating lasting change.
        </span>
        Learning fast and scaling excellence
      </h2>
      <p
        class="content__paragraph"
      >
        The results of all of this mean that with less time waiting and more time browsing, the Fortnum’s Tea Experience has delivered an increase in converted sales and web traffic as well as a huge decrease in server times from 3 to 0.6 seconds.
      </p>
      <div
        class="Card listBoxContainer"
      >
        <div
          class="listBox listBox"
        >
          <h3
            class="listBox__heading"
          >
            Since launch in March 2017, the Tea Experience has delivered
          </h3>
          <ul>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Increase in new customers vs last year
              </div>
              <div>
                +11%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Increase in existing customers vs last year
              </div>
              <div>
                +33%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Increase in conversion rate
              </div>
              <div>
                +8%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Increase in average order value (AOV)
              </div>
              <div>
                +39%
              </div>
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div>
                Improvement in average page load times
              </div>
              <div>
                +21%
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  <div
    class="whatNext"
  >
    <div
      class="whatNext__tilesContainer"
    >
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/pride/"
          >
            <img
              alt="Pride in London"
              class="whatNext__image"
              src="pride.gif"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Pride in London
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/pride/"
        >
          <h2
            class="whatNext__title"
          >
            Delivering a best-in-class digital experience
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/financial-times/"
          >
            <img
              alt="Financial Times"
              class="whatNext__image"
              src="financialTimesReadMore.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Financial Times
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/financial-times/"
        >
          <h2
            class="whatNext__title"
          >
            Lasting change for a media giant
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/car-trawler/"
          >
            <img
              alt="Car Trawler"
              class="whatNext__image"
              src="carTrawler.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Car Trawler
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/car-trawler/"
        >
          <h2
            class="whatNext__title"
          >
            Creating complete CMS control for travel brands
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
    </div>
    <a
      class="whatNext__button"
      href="/our-work/"
    >
      See more of our work
    </a>
  </div>,
  <section
    class="contactUsContainer"
    data-cy="contact-us-slice"
    id="contactUs"
  >
    <h2
      class="header"
    >
      We can help you
    </h2>
    <div
      class="contentContainer"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          Create & validate new ideas
        </li>
        <li
          class="item"
        >
          Deliver great quality products & services, fast
        </li>
        <li
          class="item"
        >
          Be bold with technology
        </li>
        <li
          class="item"
        >
          Be more customer centric
        </li>
        <li
          class="item"
        >
          Improve efficiency with lean practices
        </li>
        <li
          class="item"
        >
          Build capability & confidence
        </li>
      </ul>
      <div
        class="imgContainer"
      />
    </div>
    <div>
      <a
        class="mailToLink"
        data-cy="email-link"
        href="mailto:hello@red-badger.com?Subject=Can%20you%20help%20me%20with%20..."
      >
        Send an email
      </a>
      <span
        class="talkToUs"
      >
        hello@red-badger.com
      </span>
    </div>
  </section>,
]
`;
